MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -1232,184 +1234,194 @@ GetOverlappingMemorySpaceRegion (
+  @param[in]      DescriptorSize                  Size, in bytes, of each descriptor region in the array
+  @param[in]      MemorySpaceMapDescriptorCount   Number of entries in the MemorySpaceMap array
+  @param[in]      MemorySpaceMap                  The system memory space map (GCD memory map)
+  @param[in]      MemorySpaceMapDescriptorSize    Size, in bytes, of each descriptor region in the memory space
+                                                  map array
+  @param[in]      DetermineSize                   If TRUE, then this function will only determine the required
+                                                  buffer size for the output memory map. If FALSE, then this
+                                                  function will fill in the memory map
 
   @retval EFI_SUCCESS                   Successfully filled in the memory map
-  @retval EFI_OUT_OF_RECOURCES          Failed to allocate pools
-  @retval EFI_INVALID_PARAMETER         MemoryMap == NULL, *MemoryMap == NULL, *MemoryMapSize == 0, or
-                                        DescriptorSize == 0
+  @retval EFI_INVALID_PARAMETER         An input parameter was invalid.
 **/
 EFI_STATUS
 FillInMemoryMap (
   IN OUT    UINTN                            *MemoryMapSize,
-  IN OUT    EFI_MEMORY_DESCRIPTOR            **MemoryMap,
+  IN OUT    EFI_MEMORY_DESCRIPTOR            *MemoryMap,
+  IN        UINTN                            MemoryMapBufferSize,
+  IN        EFI_MEMORY_DESCRIPTOR            *InsertionPoint,
   IN CONST  UINTN                            *DescriptorSize,
   IN CONST  UINTN                            *MemorySpaceMapDescriptorCount,
   IN        EFI_GCD_MEMORY_SPACE_DESCRIPTOR  *MemorySpaceMap,
-  IN CONST  UINTN                            *MemorySpaceMapDescriptorSize
+  IN CONST  UINTN                            *MemorySpaceMapDescriptorSize,
+  IN OUT    BOOLEAN                          DetermineSize
   )
 {
-  EFI_MEMORY_DESCRIPTOR  *OldMemoryMapCurrent, *OldMemoryMapEnd, *NewMemoryMapStart, *NewMemoryMapCurrent;
+  EFI_MEMORY_DESCRIPTOR  *MemoryMapCurrent, *MemoryMapEnd;
   EFI_PHYSICAL_ADDRESS   LastEntryEnd, NextEntryStart, StartOfAddressSpace, EndOfAddressSpace;
   EFI_GCD_MEMORY_TYPE    GcdType = 0;
   UINT64                 RemainingLength, OverlapLength;
-  UINTN                  NewMemoryMapSize, AdditionalEntriesCount, LoopIteration;
+  UINTN                  AdditionalEntriesCount;
 
-  if ((MemoryMap == NULL) || (*MemoryMap == NULL) ||
+  if ((MemoryMap == NULL) || (MemoryMap == NULL) ||
       (MemoryMapSize == NULL) || (*MemoryMapSize == 0) ||
       (*DescriptorSize == 0) || (MemorySpaceMap == NULL) ||
-      (MemorySpaceMapDescriptorSize == NULL) || (MemorySpaceMapDescriptorCount == NULL))
+      (MemorySpaceMapDescriptorSize == NULL) || (MemorySpaceMapDescriptorCount == NULL) ||
+      (!DetermineSize && (InsertionPoint == NULL)))
   {
+    DEBUG ((DEBUG_ERROR, "%a - Function had NULL input(s)!\n", __FUNCTION__));

```

Comment:
For new additions, we should go ahead and use `__func__`.

```suggestion
    DEBUG ((DEBUG_ERROR, "%a - Function had NULL input(s)!\n", __func__));
```